### PR TITLE
Task/cleanup url utils

### PR DESCRIPTION
**Pull request by sophistifunk**
# Clean up url utils

* Separate the "specific url building" functions from the "general url manipulation" functions
* Coalesce duplicate methods, references and tests
* Remove dead code
* Tweak some names and signatures
* Additional documentation and unit tests
* General related clean-up
* Convert UrlUtils and UrlBuilder modules into namespaces instead of exporting all their methods at the package level
* Move UrlUtils and UrlBuilder to TypeScript

### CODE REVIEW

**Comment by sophistifunk** [comment]
dontlook

**Comment by kzantow** [comment]
so excited for this to land!! :)

**Comment by sophistifunk** [comment]
Last two builds were green, this is a rebase onto master, but dev is done, ready to be scrutinised :D

**Comment by scherler** [comment]
LGTM

File: blueocean-dashboard/reports/junit.xml [hunk 1]
```diff
@@ -8,11 +8,11 @@
     </testcase>
```
Comment (kzantow) [inline blueocean-dashboard/reports/junit.xml]: why is this file in the repo? (i see it wasn't _added_)

File: blueocean-dashboard/src/main/js/components/ActivityDetailsRow.jsx [hunk 2]
```diff
@@ -31,8 +31,8 @@ class ActivityDetailsRow extends Component {
         }
 
         const resultRun = run.result === 'UNKNOWN' ? run.state : run.result;
-        const runDetailsUrl = buildRunDetailsUrl(pipeline.organization, pipeline.fullName, decodeURIComponent(run.pipeline), run.id, 'pipeline');
-        const changesUrl = buildRunDetailsUrl(pipeline.organization, pipeline.fullName, decodeURIComponent(run.pipeline), run.id, 'changes');
+        const runDetailsUrl = UrlBuilder.buildRunUrl(pipeline.organization, pipeline.fullName, decodeURIComponent(run.pipeline), run.id, 'pipeline');
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/components/ActivityDetailsRow.jsx]: should `buildRunUrl` still be called `buildRunDetailsUrl`?

File: blueocean-dashboard/src/main/js/components/BranchDetailsRow.jsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ import Extensions from '@jenkins-cd/js-extensions';
 import { observer } from 'mobx-react';
 
 import RunMessageCell from './RunMessageCell';
-import { buildRunDetailsUrl } from '../util/UrlUtils';
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/components/BranchDetailsRow.jsx]: 🎉

File: blueocean-dashboard/src/main/js/components/karaoke/components/Pipeline.jsx [hunk 4]
```diff
@@ -27,7 +39,7 @@ export default class Pipeline extends Component {
         this.listener = {};
         this.sseEventHandler = this.sseEventHandler.bind(this);
         // query parameter before preference
-        this.classicLog = calculateLogView(props) || KaraokeConfig.getPreference('runDetails.logView').value === 'classic';
+        this.classicLog = isClassicLogView(props) || KaraokeConfig.getPreference('runDetails.logView').value === 'classic';
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/components/karaoke/components/Pipeline.jsx]: 🎉

File: blueocean-dashboard/reports/junit.xml (hunk 1 above)
Comment (sophistifunk) [reply blueocean-dashboard/reports/junit.xml]: Ew.

File: blueocean-core-js/src/js/UrlBuilder.ts [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+
+    return buildRunUrl(organizationName, pipelineFullName, detailName, runId);
+}
+
+/**
+ * Builds a run details view url from identifiers
+ */
+export function buildRunUrl(organizationName, pipelineFullName, branchOrPipelineName, runId, tabName = 'pipeline') {
+    const baseUrl =
+        `/organizations/${encodeURIComponent(organizationName)}` +
```
Comment (kzantow) [inline blueocean-core-js/src/js/UrlBuilder.ts]: NIT: either `+` or backtick style, not both

Comment (sophistifunk) [reply blueocean-core-js/src/js/UrlBuilder.ts]: It's just too long to read all in one, no need to break up the individual "section" templates IMO

**Comment by sophistifunk** [comment]
Righto, it's green so I'm gonna merge it and get back to git auth. We can easily address the names of the run url functions in the future since they're only exported from one place now and have unique names :D

**Pull request merged**